### PR TITLE
feat: Replace Animation Library to Reanimated

### PR DESCRIPTION
**Pull request by goheroes**
Apply Reanimated for UI/UX experiences

### CODE REVIEW

**Comment by LunatiqueCoder** [comment]
Ohhh this is actually so nice! Can't wait to take a better look at it. Hopefully it will be this weekend or the next. 🙈

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "react-native": "npm:react-native-tvos@0.68.2-5",
     "react-native-builder-bob": "^0.18.2",
     "react-native-video": "^5.2.0",
+    "react-native-reanimated": "^2.12.0",
```
Comment (LunatiqueCoder) [inline package.json:45]: We will need `react-native-reanimated` as a peer dependency as well. 🙏

**Review by LunatiqueCoder — COMMENTED** [review]
I didn't test the changes yet, nor did I have a closer look at the code, but I'm wondering how much effort would require to keep both implementations by having a `withReanimated: boolean` prop. 🤔 I think that letting the users choose between the two it's the best way to go, of course only if that's possible.

**Comment by goheroes** [comment]
> I didn't test the changes yet, nor did I have a closer look at the code, but I'm wondering how much effort would require to keep both implementations by having a `withReanimated: boolean` prop. 🤔 I think that letting the users choose between the two it's the best way to go, of course only if that's possible.

I completely agree with you and hope so.
However, implementing animation effects, including user interaction, while the video is running, is definitely not feasible on a traditional single thread, especially when playing heavy video with high quality.
I think it would be better to aim to avoid bottlenecks through worklets.

**Comment by LunatiqueCoder** [comment]
Will test the changes, then I will try to merge everything together so it can play nicely without `reanimated`

**Comment by LunatiqueCoder** [comment]
Tested, changes look good. Trying to find a way to implement a `reanimated: boolean` prop for this.

**Comment by LunatiqueCoder** [comment]
Merging this for now, however, I think I will extract your logic in a separate package and let the user decide if what he wants to use it for the animations.

**Comment by LunatiqueCoder** [comment]
Merging this for now, however, I think I will extract your logic in a separate package and let the user decide if what he wants to use it for the animations.

**Review by LunatiqueCoder — APPROVED** [review]

**Comment by LunatiqueCoder** [comment]
@goheroes I extracted your logic into `@react-native-media-console/reanimated`. Be sure to check the docs and see how to use it. There should be no breaking changes.

**Pull request merged**